pkg/gadgets/audit/seccomp/tracer/tracer.go

```diff
@@ -172,3 +175,83 @@ func (t *Tracer) Close() {
 	}
 	t.objs.Close()
 }
+
+// ---
+
+func (t *Tracer) Start() error {
+	var err error
+	var spec *ebpf.CollectionSpec
+
+	config := t.config
+
+	spec, err = loadAuditseccomp()
+	if err != nil {
+		return fmt.Errorf("failed to load ebpf program: %w", err)
+	}
+
+	mapReplacements := map[string]*ebpf.Map{}
+	filterByMntNs := false
+
+	if config.MountnsMap != nil {
+		filterByMntNs = true
+		mapReplacements["mount_ns_filter"] = config.MountnsMap
+	}
+	if config.ContainersMap != nil {
+		mapReplacements["containers"] = config.ContainersMap
+	}
+
+	consts := map[string]interface{}{
+		"filter_by_mnt_ns": filterByMntNs,
+	}
+	if err := spec.RewriteConstants(consts); err != nil {
+		return fmt.Errorf("error RewriteConstants: %w", err)
+	}
+
+	opts := ebpf.CollectionOptions{
+		MapReplacements: mapReplacements,
+	}
+
+	if err := spec.LoadAndAssign(&t.objs, &opts); err != nil {
+		return fmt.Errorf("failed to load ebpf program: %w", err)
+	}
+
+	t.reader, err = perf.NewReader(t.objs.Events, gadgets.PerfBufferPages*os.Getpagesize())
+	if err != nil {
+		return fmt.Errorf("failed to get a perf reader: %w", err)
+	}
+
+	t.progLink, err = link.Kprobe("audit_seccomp", t.objs.IgAuditSecc, nil)
+	if err != nil {
+		return fmt.Errorf("failed to attach kprobe: %w", err)
+	}
+
+	go t.run()
+
+	return nil
```

Comment:
Does the following work?

```suggestion
	if err := t.start(); err != nil {
		t.Stop()
		return err
	}
	return nil
```